repondeur/zam_repondeur/views/lectures.py

```diff
@@ -1,94 +1,65 @@
 
-
-@dataclass
-class Lecture:
-    chambre: str
-    session: str
-    num_texte: str
-
-    @property
-    def chambre_disp(self) -> str:
-        return CHAMBRES[self.chambre]
-
-    def __str__(self) -> str:
-        return f"{self.chambre_disp}, session {self.session}, texte nº {self.num_texte}"
-
-    def __lt__(self, other: Any) -> bool:
-        if type(self) != type(other):
-            return NotImplemented
-        return (self.chambre, self.session, int(self.num_texte)) < (
-            other.chambre,
-            other.session,
-            int(other.num_texte),
-        )
+CURRENT_LEGISLATURE = 15
 
 
 @view_config(route_name="lectures_list", renderer="lectures_list.html")
 def lectures_list(request: Request) -> dict:
-    return {
-        "lectures": list(
-            sorted(
-                list_lectures(request.registry.settings["zam.data_dir"]), reverse=True
-            )
-        )
-    }
-
-
-def list_lectures(dirname: str) -> Generator:
-    if not os.path.isdir(dirname):
-        return
-    for name in os.listdir(dirname):
-        mo = RE_TEXTE.match(name)
-        if mo is not None:
-            yield Lecture(**mo.groupdict())  # type: ignore
+    return {"lectures": Lecture.all()}
 
 
 @view_defaults(route_name="lectures_add", renderer="lectures_add.html")
 class LecturesAdd:
     def __init__(self, request: Request) -> None:
         self.request = request
-        self.data_dir = self.request.registry.settings["zam.data_dir"]
+        self.dossiers_by_uid = get_dossiers_legislatifs(legislature=CURRENT_LEGISLATURE)
+        self.lectures_by_dossier = {
+            dossier.uid: {
+                lecture.texte.uid: f"{lecture.chambre} : {lecture.titre} (texte nº {lecture.texte.numero} déposé le {lecture.texte.date_depot.strftime('%d/%m/%Y')})"  # noqa
+                for lecture in dossier.lectures.values()
+            }
+            for dossier in self.dossiers_by_uid.values()
+        }
 
     @view_config(request_method="GET")
     def get(self) -> dict:
-        return self._form_data()
+        return {
+            "dossiers": list(self.dossiers_by_uid.values()),
+            "lectures_by_dossier": self.lectures_by_dossier,
+        }
 
     @view_config(request_method="POST")
     def post(self) -> Response:
-        chambre = self.request.POST["chambre"]
-        session = self.request.POST["session"]
-        num_texte = self.request.POST["num_texte"]
-        if chambre not in CHAMBRES:
-            raise HTTPBadRequest
-
-        lecture_dir = os.path.join(self.data_dir, f"{chambre}-{session}-{num_texte}")
-        if os.path.isdir(lecture_dir):
+        dossier_uid = self.request.POST["dossier"]
+        texte_uid = self.request.POST["lecture"]
+
+        dossier = self.dossiers_by_uid[dossier_uid]
+        lecture = dossier.lectures[texte_uid]
+
+        if lecture.chambre == Chambre.AN:
+            chambre = "an"
```

Comment:
I don't get why you don't reuse directly `lecture.chambre`?

Reply:
You're right. Changed in 034a583d9e03f9a0bb22bc289a85edc3cf5cbba6.